### PR TITLE
Types

### PR DESCRIPTION
**Pull request by BilligsterUser**
#2 

#5

### CODE REVIEW

File: src/CashuMint.ts [hunk 1]
```diff
@@ -9,40 +10,37 @@ class CashuMint {
...
+    async requestMint(amount: number): Promise<{ pr: string, hash: string }> {
         const { data } = await axios.get<{ pr: string, hash: string }>(`${this.mintUrl}/mint`, {
             params: { amount }
         })
         return data
     }
-    async mint(payloads: string, paymentHash = '') {
-        const { data } = await axios.post(`${this.mintUrl}/mint`, payloads,
+    async mint(payloads: { outputs: Array<{ amount: number, B_: string }> }, paymentHash = "") {
+        const { data } = await axios.post<{ promises: Array<SerealizedBlindedSignature> | { error: string } }>(`${this.mintUrl}/mint`, payloads,
```
[the remark this answers is not in the document]
Comment (gandlafbtc) [reply src/CashuMint.ts:31]: i figured it out down the line. we can merge this change. :+1:

**Review by gandlafbtc — COMMENTED** [review]
Godsent PR :rocket: 

Only a few comments

File: src/utils.ts [hunk 2]
```diff
@@ -30,26 +31,24 @@ function bigIntStringify(key: any, value: any) {
...
+function getDecodedProofs(token: string): Token {
+    const obj = encodeBase64ToJson<Token | Array<Proof>>(token)
+
+    if (Array.isArray(obj)) {
+        return { proofs: obj, mints: [] }
     }
-        const proofs: Array<Proof> = obj
-        return { proofs, mints: [] }
+    // check if v2
+    return { proofs: obj.proofs, mints: obj?.mints ?? [] }
```
[the remark this answers is not in the document]
Comment (BilligsterUser) [reply src/utils.ts:49]: done

File: src/CashuWallet.ts [hunk 3]
```diff
@@ -114,10 +114,9 @@ class CashuWallet {
 
     async requestTokens(amount: number, hash: string): Promise<Array<Proof>> {
         const { blindedMessages, secrets, rs } = await this.createRandomBlindedMessages(amount)
-        const payloads: { outputs: Array<{ amount: number, B_: string }> } = { outputs: blindedMessages }
-        const payloadsJson = JSON.parse(JSON.stringify({ payloads }, bigIntStringify))
-        const { promises } = await this.mint.mint(payloadsJson.payloads, hash)
-        if (promises.error) {
+        const payloads = { outputs: blindedMessages }
+        const { promises } = await this.mint.mint(payloads, hash)
+        if ('error' in promises) {
```
[the remark this answers is not in the document]
Comment (BilligsterUser) [reply src/CashuWallet.ts:119]: ok

File: src/CashuWallet.ts [hunk 4]
```diff
@@ -32,15 +32,15 @@ class CashuWallet {
...
         }
         const { spendable } = await this.mint.check(payload)
         //return only the proofs that are NOT spendable
-        return proofs.filter((p,i)=>!spendable[i])
+        return proofs.filter((_, i) => !spendable[i])
     }
 
-    async requestMint(amount: number) {
-        return await this.mint.requestMint(amount)
+    requestMint(amount: number) {
```
[the remark this answers is not in the document]
Comment (BilligsterUser) [reply src/CashuWallet.ts:42]: no this.mint.requestMint(amount) still returns a promis

**Review by gandlafbtc — APPROVED** [review]
:+1:

**Pull request merged**